kartothek/io_components/utils.py

```diff
@@ -341,30 +340,34 @@ def sort_values_categorical(df, column):
     return df.sort_values(by=[column]).reset_index(drop=True)
 
 
-def check_single_table_dataset(dataset, expected_table=None):
+def check_single_table_dataset(
+    dataset: Union[DatasetFactory, DatasetMetadata], expected_table=None
+):
     """
     Raise if the given dataset is not a single-table dataset.
 
     Parameters
     ----------
-    dataset: kartothek.core.dataset.DatasetMetadata
+    dataset: kartothek.core.factory.DatasetFactory
         The dataset to be validated
     expected_table: Optional[str]
         Ensure that the table in the dataset is the same as the given one.
     """
-
-    if len(dataset.tables) > 1:
-        raise TypeError(
-            "Expected single table dataset but found dataset with tables: `{}`".format(
-                dataset.tables
+    if (isinstance(dataset, DatasetFactory) and dataset.exists) or isinstance(
```

Comment:
This additional guard wasn't there before. Is there a test for this behavior?

Reply:
The function `check_single_table_dataset` didn't have any unit tests. I added some